src/gui/transferlistwidget.cpp

```diff
@@ -588,8 +588,10 @@ void TransferListWidget::openSelectedTorrentsFolder() const
 void TransferListWidget::previewSelectedTorrents()
 {
     foreach (BitTorrent::TorrentHandle *const torrent, getSelectedTorrents()) {
-        if (torrent->hasMetadata())
+        if (torrent->isPreviewable())
             new PreviewSelectDialog(this, torrent);
```

Comment:
IMO, calling "show()" in dialog constructor is really bad idea.

Reply:
It is.